Tests/DatadogTests/Datadog/Mocks/DatadogInternal/PassthroughCoreMock.swift

```diff
@@ -68,6 +68,8 @@ internal final class PassthroughCoreMock: DatadogV1CoreProtocol, FeatureScope {
         self.expectation = expectation
         self.bypassConsentExpectation = bypassConsentExpectation
         self.messageReceiver = messageReceiver
+
+        messageReceiver.receive(message: .context(context), from: self)
```

Comment:
It feels a bit that we're re-implementing `DatadogCore` in `PassthroughCoreMock`, no? Step by step, it gains more behaviours similar to real core. I wonder if this trend will continue as we grow V2 adoption - any take on this @maxep ?

TBC: I love the idea of `PassthroughCoreMock` that isolates everything from real storage - just trying to anticipate the future.

Reply:
I agree! Sending the context through the bus at registration shouldn't be a requirement/convention of the core. I think that this should be enforced by the API.
Another option would be to add a `core.context { _ in }` async method to the core so we can get the context at anytime. But then it kindof questioned the concept of `FeatureScope` where the context is given with the writer.

We could also split `eventWriteContext` with the following API:
```swift
protocol DatadogCoreProtocol {

    func register(feature: DatadogFeature) throws -> Writer

    func register(integration: DatadogFeatureIntegration) throws

    func context(_ block: (DatadogContext) -> Void)
```

I won't be doing that in this PR, but that something we can discuss!

Reply:
> Another option would be to add a `core.context { _ in }` async method to the core so we can get the context at anytime. But then it kindof questioned the concept of `FeatureScope` where the context is given with the writer.

To me both are not mutually exclusive. There is a difference between "get context" and "write event" behaviours (where the latter is supplied with `context` for write convenience). 

Anyway, I'm just putting a mark on the topic of core mocks - happy to know we're both aware of its evolution 👍. And yes, we can discuss it at some point to see what can be improved.